### ISSUE TITLE
Uncharger not behaving on some zwitterions.

### ISSUE THREAD

**Issue by greglandrum** [rdkit/rdkit#2451]
**Description:**
```
# this one is fine:
In [9]: Chem.MolToSmiles(uncharger.uncharge(Chem.MolFromSmiles('C[NH+](C)(C)')))                                                                 
[15:09:59] Running Uncharger
[15:09:59] Removed positive charge.
Out[9]: 'CN(C)C'

# this one is not:
In [10]: Chem.MolToSmiles(uncharger.uncharge(Chem.MolFromSmiles('C[NH+](C)([O-])')))                                                             
[15:10:09] Running Uncharger
[15:10:09] Removed negative charge.
Out[10]: 'C[NH+](C)O'
```

The pattern that's currently being used explicitly rules out the `[NH+]` since it's bonded to a negatively charged atom, but allows the `[O-]` to be protonated. That doesn't seem right.

**Comment by greglandrum** [comment]
@mcs07 : I think this behavior came over from MolVS.
Am I misunderstanding something when I call it a bug?

**Comment by mcs07** [comment]
Yeah, I think this is (another) bug that is a consequence of the slightly bizarre initial implementation of the Uncharger in MolVS. Back when you did some fixes in #2374, I also did a more substantial overhaul in MolVS: https://github.com/mcs07/MolVS/commit/2bfee11768cf32932b5572684a8e36e74336f8e3

Basically I got rid of the weird SMARTS patterns and instead just look at all charges, but with a list of patterns to skip (currently just charge-separated nitro group). It seems to work for your examples above.
